packages/cardano-services-client/src/WebSocket.ts

```diff
@@ -50,10 +76,44 @@ interface EpochRollover {
   protocolParameters: Cardano.ProtocolParameters;
 }
 
+const deserializeDatum = (tx: Cardano.HydratedTx) => {
+  for (const output of tx.body.outputs)
+    if (output.datum) output.datum = Serialization.PlutusData.fromCbor(output.datum as unknown as HexBlob).toCore();
+};
+
+const deserializeMetadata = (tx: Cardano.HydratedTx) => {
+  if (tx.auxiliaryData)
+    tx.auxiliaryData = {
+      blob: new Map(
+        (tx.auxiliaryData as unknown as MetadataDbModel[]).map((metadata) => {
+          const bKey = BigInt(metadata[0]);
+
+          return [bKey, Serialization.GeneralTransactionMetadata.fromCbor(metadata[1]).toCore().get(bKey)!] as const;
+        })
+      )
+    };
+};
+
 const isEventError = (error: unknown): error is { error: Error } =>
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   typeof error === 'object' && !!error && (error as any).error instanceof Error;
 
+export const isTxRelevant = (
```

Comment:
Also need to check certificates

Reply:
Actually no. This is not a general purpose function, it is dedicated to `ChainHistoryProvider.transactionsByAddresses`.
This is meant to check transaction only by addresses as `DbSyncChainHistoryProvider.transactionsByAddresses` does.

Reply:
If `DbSyncChainHistoryProvider.transactionsByAddresses` then it's an edge case bug. You can make a transaction that has a certificate (e.g. delegation) without using any inputs/outputs of an associated address

Reply:
It could be, anyway fixing pre-existing bugs is OT for this PR.
The intent here is to _replicate_ what the `(DbSync|http)Provider`s do.

Reply:
Please log this bug for separate consideration @iccicci

Reply:
Created LW-11629